### PR TITLE
fix: upgrade node default runtime to 14

### PR DESCRIPTION
**Pull request by michaelbrewer**
#### Description of changes

As we approach the EOL for node 12 security updates new
lambdas should be ideally be on node 14

- [Node release schedule](https://github.com/nodejs/Release#release-schedule)

#### Description of how you validated changes

Generate example stacks to verify node 14 version was used

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [X] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/migration/node.function.test.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ describe('nodejs version migration tests', () => {
 
     fs.writeFileSync(projectConfigFileName, projectConfigContent, 'utf-8');
 
-    // Write back the nodejs version to 'node10.x' to test migration
+    // Write back the nodejs version to 'node12.x' to test migration
```
Comment (attilah) [inline packages/amplify-e2e-tests/src/__tests__/migration/node.function.test.ts:47]: Leave it on 10.x, but change `nodejs12.x` to `nodejs14.x` in line 59.

**Review by attilah — CHANGES_REQUESTED** [review]
thanks for the PR LGTM!

Except 1 change, I added a comment.

When node migration was added to the CLI, node14.x was not supported by CFN for custom lambda handlers.

File: packages/amplify-e2e-tests/src/__tests__/migration/node.function.test.ts (hunk 1 above)
Comment (michaelbrewer) [reply packages/amplify-e2e-tests/src/__tests__/migration/node.function.test.ts]: ok changes @attilah

**Review by attilah — APPROVED** [review]
LGTM

**Pull request merged**